test/widgets/inbox_test.dart

```diff
@@ -385,6 +450,50 @@ void main() {
           checkAppearsUncollapsed(tester, 1, findSectionContent);
         });
 
+        testWidgets('collapse stream section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final topicFinder = find.text('stream 1 topic 4' ).hitTestable();
+          final listFinder = find.byType(Scrollable);
+
+          // Scroll part of [_StreamSection] offscreen.
+          await dragUntilInvisible(
+            tester, topicFinder, listFinder, const Offset(0, -50));
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          // Check that the header is present (which must therefore
+          // be as a sticky header).
+          check(headerRow).isNotNull();
+
+          await tapCollapseIcon(tester, 1);
+
+          final headerRowAfterTap = findStreamHeaderRow(tester, 1);
+          // Check that the header is still visible even after
+          // collapsing the section.
+          check(headerRowAfterTap).isNotNull();
+        });
+
+        testWidgets('collapse stream section in the middle of screen', (tester) async {
+          await setupVarious(tester);
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          final rectBeforeTap = tester.getRect(find.byWidget(headerRow!));
+
+          // Check that the header is present (which must therefore
+          // be somewhere in the middle of screen)
```

Comment:
Hmm, but why does this mean it's in the middle of the screen? As far as one can see from reading this test's own code, it seems like it could just as well be the sticky header at the top of the viewport.

The reason it's not at the top is because there are unread DMs too, and so there's a DMs section above it. But that's a detail of `setupVarious`, and it seems quite possible that someone in the future could change that, causing this test to just always pass even if the functionality it's meant to test subsequently gets broken. After all, when they do make that change, nothing in this test will break.

There are two basic ways to solve that sort of problem:
* Add a check within this test that verifies that the setup meets the conditions that this test specifically needs from it.
* Specify the relevant details directly in the test, rather than far away in a place where it's not obvious that there are tests that rely on them.

For example here:
* For the first approach, check that the top of `rectBeforeTap` is somewhere below the top of the scrollable viewport.
* Or for the second approach, instead of using `setupVarious`, call `setupPage` directly. That setup can then be simpler than what `setupVarious` does, focusing only on the details this test needs. For example: one DM conversation, then a stream with just one conversation, then a second stream with lots of conversations (enough so that it's clearly more than a screenful, so that scrolling is possible).

Here I think either approach would work fine. (Only one is needed, not both.)